src/communication/makeRetrying.ts

```diff
@@ -87,7 +87,7 @@ export default function makeRetrying(axiosInstance: AxiosInstance) {
   axiosInstance.interceptors.request.use((config: AxiosRequestConfig & Partial<AttemptState>) => {
     // If the request is a POST or DELETE one and does not yet have the idempotency header, add one now.
     if (unsafeMethods.has(config.method!) && config.headers?.[idempotencyHeaderName] == undefined) {
-      Object.assign(config.headers, generateIdempotencyHeader());
+      config.headers = {...config.headers, ...generateIdempotencyHeader()}
```

Comment:
One alternative would be:
```javascript
Object.assign(config.headers ??= {}, generateIdempotencyHeader());
```
Too far-fetched?

Reply:
Actually, `headers` can actually never be `undefined` as long as default headers are provided, which we do in `NetworkClient`. [Axios' source code](//github.com/axios/axios/blob/412d3bd6078433dda4a7eb4d86e021dbc57fdeb2/lib/core/mergeConfig.js) confirms this.

Does that change our mind?

Reply:
discussed, decision to keep this version as it's the most future proof.
```Object.assign(config.headers ??= {}, generateIdempotencyHeader());```